### PR TITLE
Admin backend functionality

### PR DESCRIPTION
**Pull request by kolyaventuri**
- Admins can access backend
- Admins can modify countries and assign data
- Admins can modify currencies/languages

### CODE REVIEW

File: app/controllers/admin/country_languages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Admin::
+module Admin
+  class CountryLanguagesController < BaseController
```
Comment (kolyaventuri) [inline app/controllers/admin/country_languages_controller.rb:3]: @iandouglas @AliSchlereth So I ended up creating this class for admins adding languages to countries, and a separate languages controller for managing just language records. 

There's still the UX thing you mentioned, with altering data, clicking "Add Language" and losing any unsaved changes, but I'm going to be more focus on getting the functionality in there, even if the UX does need improvement.

File: config/routes.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
   resources :countries, only: [:show]
+  get '/travel/:country_1/:country_2', to: 'countries#render', as: 'countries_render'
```
Comment (kolyaventuri) [inline config/routes.rb:5]: @iandouglas @AliSchlereth Is this the best way to handle this? Haven't build out `CountriesController#render` yet, but if I'm trying to allow access to an endpoint such as `/travel/US/GB` (travel from US to England), is that the proper way to go about this? Or should I make it using nested resources, so it's more like `/country/US/country/GB`. The former feels nicer from a UX standpoint.

File: app/controllers/admin/country_languages_controller.rb (hunk 1 above)
Comment (AliSchlereth) [reply app/controllers/admin/country_languages_controller.rb:3]: @kolyaventuri While wrapping your controller in a module works for namespacing, I've always seen it done `Admin::CountryLanguagesController`. Putting this file in the admin sub directory is what Rails will expect from using this syntax.

File: config/routes.rb (hunk 2 above)
Comment (AliSchlereth) [reply config/routes.rb:5]: I want to think about the uri a bit more, but Do have a comment about using render as an action. Try to stick to the 7 restful actions in your controller. Try to match what you're doing to whichever action fits best, also remember you can create another controller if you already have used that action in the CountriesController.

**Pull request merged**